src/spec-configuration/containerFeaturesOrder.ts

```diff
@@ -3,133 +3,538 @@
+
+	if (!config.overrideFeatureInstallOrder) {
+		return worklist;
+	}
+
+	const overrideFeatureNodes: FNode[] = [];
+
+	// Create an artificial node for each Feature in the override property.
+	// Reverse iterate to remove out items from config that have been processed.
+	for (let j = config.overrideFeatureInstallOrder.length - 1; j >= 0; j--) {
+		const userFeatureId = config.overrideFeatureInstallOrder[j];
+		const overrideNode: FNode = {
+			type: 'artificial',
+			userFeatureId,
+			options: {},
+			dependsOn: [],
+			installsAfter: [],
+		};
+		const processed = await processFeature(overrideNode);
+		if (!processed) {
+			throw new Error(`Feature '${userFeatureId}' in 'overrideFeatureInstallOrder' could not be processed.`);
+		}
+
+		if (featureSupportsDependencies(processed)) {
+			overrideNode.featureSet = processed;
+			overrideFeatureNodes.push(overrideNode);
+			config.overrideFeatureInstallOrder.splice(j, 1);
+		}
+	}
+
+	overrideFeatureNodes.reverse();
+	output.write(`[override-feature-nodes]: ${overrideFeatureNodes.map(n => n.userFeatureId).join(', ')}`, LogLevel.Info);
+
+	for (let i = 0; i < overrideFeatureNodes.length - 1; i++) {
+		// Scan the resolvedGraph for a node that matches the next override node.
+		const currentOverrideNode = overrideFeatureNodes[i];
+		const nextOverrideNode = overrideFeatureNodes[i + 1];
+		let countMatches = 0;
+		for (const node of worklist) {
+			if (satisfiesSoftDependency(node, nextOverrideNode)) {
+				countMatches++;
+				// Add a soft dependency on the next override node.
+				output.write(`[override-feature-add-relationship]: ${node.userFeatureId} installsAfter ${currentOverrideNode.userFeatureId}`, LogLevel.Info);
+				node.installsAfter.push(currentOverrideNode);
+			}
+		}
+
+		// Invariant: Every Feature indicated in the 'overrideFeatureInstallOrder' property should have at least one corresponding Feature in the resolved graph.
+		if (!countMatches) {
+			throw new Error(`The 'overrideFeatureInstallOrder' property includes references to Feature(s) that are not queued for installation.`);
+		}
+	}
+
+	// Add the final node to each leaf node's soft-dependency adjacency list (Enforcing override Features are installed first),
+	// Ensuring that such an act does not introduce a cycle.
+	const lastOverrideNode = overrideFeatureNodes[overrideFeatureNodes.length - 1];
+	for (const node of worklist) {
+		if (node.installsAfter.length === 0 //  TODO: necessary?
+			&& node.dependsOn.length === 0 // Leaf node
+			&& !satisfiesSoftDependency(node, lastOverrideNode) // Prevent cycles with itself
+			&& overrideFeatureNodes.every(override => !satisfiesSoftDependency(node, override))) // Prevent cycles with other override nodes 
+		{
+			output.write(`[override-feature-nodes-tail]: ${node.userFeatureId} installsAfter ${lastOverrideNode.userFeatureId}`, LogLevel.Info);
+			node.installsAfter.push(lastOverrideNode);
+		}
+	}
+
+	// Return the modified worklist.
+	return worklist;
```

Comment:
So this does not include the `overrideFeatureNodes`.

Reply:
Every Feature indicated in `overrideFeatureNodes` should already be present in the `worklist` (That's the goal of [this invariant check](https://github.com/devcontainers/cli/pull/530/files#diff-5d796ceb58c276912ce5289b37ea9eda883f4abbf9b6cc1252661deccae91e67R237-R240), although I think that check needs some work).  The work done above this return is adding `installsAfter` references to the nodes in the `worklist` to influence install order in the next stage.

Reply:
This has been removed in favor of a priority-based approach evaluated per-round. See https://github.com/devcontainers/cli/commit/c8bfd1b40c50603da78d9066c80f10e3669d158c